### PR TITLE
Add tryRethrow

### PR DESCRIPTION
**Pull request by natefaubion**

### CODE REVIEW

File: src/Text/Parsing/Parser.purs [hunk 1]
```diff
@@ -123,8 +125,14 @@ consume :: forall s m. Monad m => ParserT s m Unit
...
 -- | Fail with a message.
 fail :: forall m s a. Monad m => String -> ParserT s m a
-fail message = do
-  position <- gets \(ParseState _ pos _) -> pos
-  throwError (ParseError message position)
+fail message = throwError <<< ParseError message =<< position
+
+-- | Fail with a message and a position.
+failWithPosition :: forall m s a. Monad m => String -> Position -> ParserT s m a
+failWithPosition message position = throwError (ParseError message position)
```
Comment (garyb) [inline src/Text/Parsing/Parser.purs]: Did you intend to use this in `fail` or something? Currently seems unused within the library at least.

Comment (natefaubion) [inline src/Text/Parsing/Parser.purs]: Oops, yeah :)

**Comment by natefaubion** [comment]
One other option may be to have a separate `rethrow` combinator, but it just results in more binds (which is why I baked in the try logic).

**Comment by paf31** [comment]
👍 LGTM, thanks

**Pull request merged**